x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/details_page/components/datasources/datasources_table.tsx

```diff
@@ -0,0 +1,307 @@
+  // the list of options that will be used in the filters dropdowns
+  const [datasources, namespaces, inputTypes] = useMemo((): [
+    InMemoryDatasource[],
+    FilterOption[],
+    FilterOption[]
+  ] => {
+    const namespacesValues: string[] = [];
+    const inputTypesValues: string[] = [];
+    const mappedDatasources = originalDatasources.map<InMemoryDatasource>(datasource => {
+      if (datasource.namespace && !namespacesValues.includes(datasource.namespace)) {
+        namespacesValues.push(datasource.namespace);
+      }
+
+      const dsInputTypes: string[] = [];
+      const streams = datasource.inputs.reduce(
+        (streamSummary, input) => {
+          if (!inputTypesValues.includes(input.type)) {
+            inputTypesValues.push(input.type);
+          }
+          if (!dsInputTypes.includes(input.type)) {
+            dsInputTypes.push(input.type);
+          }
+
+          streamSummary.total += input.streams.length;
+          streamSummary.enabled += input.enabled
+            ? input.streams.filter(stream => stream.enabled).length
+            : 0;
+
+          return streamSummary;
+        },
+        { total: 0, enabled: 0 }
+      );
+
+      dsInputTypes.sort(stringSortAscending);
+
+      return {
+        ...datasource,
+        streams,
+        inputTypes: dsInputTypes,
+        packageName: datasource.package?.name ?? '',
+        packageTitle: datasource.package?.title ?? '',
+        packageVersion: datasource.package?.version ?? '',
+      };
+    });
+
+    namespacesValues.sort(stringSortAscending);
+    inputTypesValues.sort(stringSortAscending);
+
+    return [
+      mappedDatasources,
+      namespacesValues.map(toFilterOption),
+      inputTypesValues.map(toFilterOption),
+    ];
+  }, [originalDatasources]);
+
+  const columns = useMemo(
+    (): EuiInMemoryTableProps<InMemoryDatasource>['columns'] => [
+      {
+        field: 'name',
+        name: i18n.translate('xpack.ingestManager.configDetails.datasourcesTable.nameColumnTitle', {
+          defaultMessage: 'Data source',
+        }),
+      },
+      {
+        field: 'description',
+        name: i18n.translate(
+          'xpack.ingestManager.configDetails.datasourcesTable.descriptionColumnTitle',
+          {
+            defaultMessage: 'Description',
+          }
+        ),
+        truncateText: true,
+      },
+      {
+        field: 'packageTitle',
+        name: i18n.translate(
+          'xpack.ingestManager.configDetails.datasourcesTable.packageNameColumnTitle',
+          {
+            defaultMessage: 'Package',
+          }
+        ),
+        render(packageTitle: string, datasource: InMemoryDatasource) {
+          return (
+            <>
+              {datasource.package && (
+                <PackageIcon
+                  packageName={datasource.package.name}
+                  version={datasource.package.version}
+                  size="m"
+                  style={{ marginRight: '1ch' }}
```

Comment:
Instead of adding inline style, we could use flex which will provide gutter and vertical centering
```tsx
<EuiFlexGroup gutter="s" alignItems="center">
  {datasource.package && (<EuiFlexItem grow={false}><PackageIcon ... /></EuiFlexItem>)}
  <EuiFlexItem grow={false}>
    {packageTitle}
  </EuiFlexItem>
</EuiFlexGroup>
```

Reply:
Nice - will make the change